p2p/gater.go

```diff
@@ -28,8 +60,11 @@ func (g *gater) InterceptPeerDial(pid peer.ID) bool {
 	return len(g.h.Network().Peers()) <= g.outbound
 }
 
-func (*gater) InterceptAddrDial(pid peer.ID, m multiaddr.Multiaddr) bool {
-	return true
+func (g *gater) InterceptAddrDial(pid peer.ID, m multiaddr.Multiaddr) bool {
+	if !g.InterceptPeerDial(pid) {
+		return false
+	}
+	return !g.isBlocked(m)
```

Comment:
nit: double-negative is always confusing to me.
would be slightly more readable for me if it's return `g.allowed(m)`